project/app/user.py

```diff
@@ -487,15 +512,45 @@ def future_budget(self, monthly_savings_goal=50, num_months=6, weighted=True):
         disc_dict_reversed = {}
         for k, v in disc_dict.items():
             disc_dict_reversed[v] = k
+        
+        """
+        if number of months < 6:
+            flag Warning
+        
+        if num transaction < 100:
+            flag WARNING
+
+        if len of discrectionary == 0:
+            remove_category = some non-disc cat
+            flag WARNING
+
+        else:
+            remove_category = max(disc_dict_reversed.items())
+        """
+
+        if len(disc_dict_reversed) == 0:
+            warning.append(f"Cannot find a discretionary category. This is likely because of insufficient transaction history.")
+            return json.dumps([avg_cat_spending_dict, warning])
+
 
         # find the key:value pair that shows which discretionary category the user
         # . spent the most money in
         max_cat = max(disc_dict_reversed.items())
 
+        if max_cat[0] < monthly_savings_goal:
+            warning = f"Your monthly savings goal of {monthly_savings_goal} is too large compared to your discretionary budget. Please input a smaller savings goal"
+            return json.dumps([None, warning])
+
         # subtract the monthly savings goal from that category
         avg_cat_spending_dict[max_cat[1]] -= monthly_savings_goal
 
-        self.past_months = num_months
+        """
```

Comment:
Delete this comment block.

Replace with:
# If warning list is not empty, add it to the return body